packages/patternfly-4/react-virtualized-extension/src/components/Virtualized/VirtualGrid.js

```diff
@@ -0,0 +1,1477 @@
+    const totalRowsHeight = instanceProps.rowSizeAndPositionManager.getTotalSize();
+
+    // Force browser to hide scrollbars when we know they aren't necessary.
+    // Otherwise once scrollbars appear they may not disappear again.
+    // For more info see issue #116
+    const verticalScrollBarSize = totalRowsHeight > height ? instanceProps.scrollbarSize : 0;
+    const horizontalScrollBarSize = totalColumnsWidth > width ? instanceProps.scrollbarSize : 0;
+
+    if (
+      horizontalScrollBarSize !== this._horizontalScrollBarSize ||
+      verticalScrollBarSize !== this._verticalScrollBarSize
+    ) {
+      this._horizontalScrollBarSize = horizontalScrollBarSize;
+      this._verticalScrollBarSize = verticalScrollBarSize;
+      this._scrollbarPresenceChanged = true;
+    }
+
+    // Also explicitly init styles to 'auto' if scrollbars are required.
+    // This works around an obscure edge case where external CSS styles have not yet been loaded,
+    // But an initial scroll index of offset is set as an external prop.
+    // Without this style, VirtualGrid would render the correct range of cells but would NOT update its internal offset.
+    // This was originally reported via clauderic/react-infinite-calendar/issues/23
+    gridStyle.overflowX = totalColumnsWidth + verticalScrollBarSize <= width ? 'hidden' : 'auto';
+    gridStyle.overflowY = totalRowsHeight + horizontalScrollBarSize <= height ? 'hidden' : 'auto';
+
+    const childrenToDisplay = this._childrenToDisplay;
+
+    const showNoContentRenderer = childrenToDisplay.length === 0 && height > 0 && width > 0;
+
+    const scrollContainerProps = {
+      ...containerProps,
+      ref: this._setScrollingContainerRef,
+      'aria-label': this.props['aria-label'],
+      'aria-rowcount': this.props['aria-rowcount'],
+      'aria-readonly': this.props['aria-readonly'],
+      className: clsx('ReactVirtualized__VirtualGrid', className),
+      id,
+      onScroll: this._onScroll,
+      role,
+      style: {
+        ...gridStyle,
+        ...style
+      },
+      tabIndex
+    };
+
+    let innerScrollContainer = null;
+    if (childrenToDisplay.length > 0) {
+      const innerScrollContainerProps = {
+        className: 'ReactVirtualized__VirtualGrid__innerScrollContainer',
+        role: containerRole,
+        style: {
+          width: autoContainerWidth ? 'auto' : totalColumnsWidth,
+          height: totalRowsHeight,
+          maxWidth: totalColumnsWidth,
+          maxHeight: totalRowsHeight,
+          overflow: 'hidden',
+          pointerEvents: isScrolling ? 'none' : '',
+          position: 'relative',
+          display: 'block',
+          ...containerStyle
+        }
+      };
+      innerScrollContainer = React.createElement(
+        innerScrollContainerComponent,
+        innerScrollContainerProps,
+        childrenToDisplay
+      );
+    }
+    return React.createElement(scrollContainerComponent, scrollContainerProps, [
```

Comment:
I'll open an upstream PR to [react-virtualized](https://github.com/bvaughn/react-virtualized) for this piece... the consumer should be able to override the default scroll elements rendered.

Reply:
this would also add support for the implementation of a virtual scrolling Data List in the future...

Reply:
i've opened [#1377](https://github.com/bvaughn/react-virtualized/pull/1377) upstream in react-virtualized. This would simplify the API consumption there quite a bit if merged.